README

------------------------------------------------------
	Requirements
------------------------------------------------------

Client Side:

	- Firefox - http://www.getfirefox.com/
	- Firebug - http://www.getfirebug.com/
	- FirePHP - http://www.firephp.org/

Server Side:
	
	- PHP 5 (complete functionality)
	- PHP 4 (most functionality)


------------------------------------------------------
	Install Tutorial
------------------------------------------------------
	
	http://www.firephp.org/HQ/Install.htm
	
	
------------------------------------------------------
	Support
------------------------------------------------------
	
	http://forum.firephp.org/
	


CHANGELOG

2009-05-01 - Release Version: 0.3.rc.1

  - (Issue 90) PHP4 compatible version of FirePHPCore
  - (Issue 98) Thrown exceptions don't send an HTTP 500 if the FirePHP exception handler is enabled
  - (Issue 85) Support associative arrays in encodeTable method in FirePHP.class.php
  - (Issue 66) Add a new getOptions() public method in API
  - (Issue 82) Define $this->options outside of __construct
  - (Issue 72) Message error if group name is null
  - (Issue 68) registerErrorHandler() and registerExceptionHandler() should returns previous handlers defined
  - (Issue 69) Add the missing register handler in the triumvirate (error, exception, assert)
  - (Issue 75) [Error & Exception Handling] Option to not exit script execution
  - (Issue 83) Exception handler can't throw exceptions
  - (Issue 80) Auto/Pre collapsing groups AND Custom group row colors

2008-11-09 - Release Version: 0.2.1

  - (Issue 70) Problem when logging resources

2008-10-21 - Release Version: 0.2.0

	- Updated version to 0.2.0
	- Switched to using __sleep instead of __wakeup
	- Added support to exclude object members when encoding
  - Add support to enable/disable logging

2008-10-17 - Release Version: 0.2.b.8
  
  - New implementation for is_utf8()
  - (Issue 55) maxObjectDepth Option not working correctly when using TABLE and EXCEPTION Type
  - Bugfix for max[Object|Array]Depth when encoding nested array/object graphs
  - Bugfix for FB::setOptions()

2008-10-16 - Release Version: 0.2.b.7

  - (Issue 45) Truncate dump when string have non utf8 cars
  - (Issue 52) logging will not work when firephp object gets stored in the session.

2008-10-16 - Release Version: 0.2.b.6

  - (Issue 37) Display file and line information for each log message
  - (Issue 51) Limit output of object graphs
  - Bugfix for encoding object members set to NULL|false|''

2008-10-14 - Release Version: 0.2.b.5

  - Updated JsonStream wildfire protocol to be more robust
  - (Issue 33) PHP error notices running demos
  - (Issue 48) Warning: ReflectionProperty::getValue() expects exactly 1 parameter, 0 given

2008-10-08 - Release Version: 0.2.b.4

	- Bugfix for logging objects with recursion

2008-10-08 - Release Version: 0.2.b.3

  - (Issue 43) Notice message in 0.2b2
  - Added support for PHP's native json_encode() if available
  - Revised object encoder to detect object recursion

2008-10-07 - Release Version: 0.2.b.2

  - (Issue 28) Need solution for logging private and protected object variables
  - Added trace() and table() aliases in FirePHP class
  - (Issue 41) Use PHP doc in FirePHP
  - (Issue 39) Static logging method for object oriented API

2008-10-01 - Release Version: 0.2.b.1

  - Added support for error and exception handling
  - Updated min PHP version for PEAR package to 5.2
  - Added version constant for library
  - Gave server library it's own wildfire plugin namespace
  - Migrated communication protocol to Wildfire JsonStream
  - Added support for console groups using "group" and "groupEnd"
  - Added support for log, info, warn and error logging aliases
  - (Issue 29) problem with TRACE when using with error_handler
  - (Issue 33) PHP error notices running demos
  - (Issue 12) undefined index php notice
  - Removed closing ?> php tags
  - (Issue 13) the code in the fb() function has a second return statement that will never be reached

2008-07-30 - Release Version: 0.1.1.3

	- Include __className property in JSON string if variable was an object
	- Bugfix - Mis-spelt "Exception" in JSON encoding code

2008-06-13 - Release Version: 0.1.1.1

	- Bugfix - Standardize windows paths in stack traces
	- Bugfix - Display correct stack trace info in windows environments
	- Bugfix - Check $_SERVER['HTTP_USER_AGENT'] before returning

2008-06-13 - Release Version: 0.1.1

	- Added support for FirePHP::TRACE log style
	- Changed license to New BSD License

2008-06-06 - Release Version: 0.0.2

  - Bugfix - Added usleep() to header writing loop to ensure unique index
  - Bugfix - Ensure chunk_split does not generate trailing "\n" with empty data header
  - Added support for FirePHP::TABLE log style
  
 

apache/Version1.php
<?php
/* ***** BEGIN LICENSE BLOCK *****
 * Version: MPL 1.1/GPL 2.0/LGPL 2.1
 *
 * The contents of this file are subject to the Mozilla Public License Version
 * 1.1 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/
 *
 * Software distributed under the License is distributed on an "AS IS" basis,
 * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
 * for the specific language governing rights and limitations under the
 * License.
 *
 * The Initial Developer of the Original Code is Christoph Dorn.
 *
 * Portions created by the Initial Developer are Copyright (C) 2006
 * the Initial Developer. All Rights Reserved.
 *
 * Contributor(s):
 *     Christoph Dorn <christoph@christophdorn.com>
 *
 * Alternatively, the contents of this file may be used under the terms of
 * either the GNU General Public License Version 2 or later (the "GPL"), or
 * the GNU Lesser General Public License Version 2.1 or later (the "LGPL"),
 * in which case the provisions of the GPL or the LGPL are applicable instead
 * of those above. If you wish to allow use of your version of this file only
 * under the terms of either the GPL or the LGPL, and not to allow others to
 * use your version of this file under the terms of the MPL, indicate your
 * decision by deleting the provisions above and replace them with the notice
 * and other provisions required by the GPL or the LGPL. If you do not delete
 * the provisions above, a recipient may use your version of this file under
 * the terms of any one of the MPL, the GPL or the LGPL.
 *
 * ***** END LICENSE BLOCK ***** */


require_once('./../Libraries/FirePHPCore/lib/FirePHPCore/fb.php');


$data = $_SERVER;
$data[] = new FirePHP();

//$data = array('VarName'=>'VarValue22','var2'=>'val2');

if($_GET['var']==1) {


/*
header('X-FirePHP-Data-100000000001:	{');
header('X-FirePHP-Data-300000000001:	"FirePHP.Firebug.Console":[');
header('X-FirePHP-Data-399999999999:	["__SKIP__"]],');
header('X-FirePHP-Data-999999999999:	"__SKIP__":"__SKIP__"}');
header('X-FirePHP-Data-368615080800:	["LOG",["PHP:$list From combien_services.php[32] getListTirage()",[{"libelle":"lundi 3 novembre 2008","date":"20081103","numtir":"2008094","DFOR":"20090103"},{"libelle":"samedi 1 novembre 2008","date":"20081101","numtir":"2008093","DFOR":"20090101"},{"libelle":"mercredi 29 octobre 2008","date":"20081029","numtir":"2008090","DFOR":"20081229"},{"libelle":"mercredi 29 octobre 2008","date":"20081029","numtir":"2008091","DFOR":"20081229"},{"libelle":"lundi 27 octobre 2008","date":"20081027","numtir":"2008089","DFOR":"20081227"},{"libelle":"lundi 20 octobre 2008","date":"20081020","numtir":"2008086","DFOR":"20081220"},{"libelle":"mercredi 15 octobre 2008","date":"20081015","numtir":"2008084","DFOR":"20081215"},{"libelle":"lundi 13 octobre 2008","date":"20081013","numtir":"2008083","DFOR":"20081213"},{"libelle":"mercredi 8 octobre 2008","date":"20081008","numtir":"2008081","DFOR":"20081208"},{"libelle":"mercredi 30 juillet 2008","date":"20080730","numtir":"2008061","DFOR":"20080929"}]]],');
*/

foreach( file('Version1Headers.txt') as $line ) {
  if( $line = trim($line) ) {
    
    $line = substr_replace($line,':',27,0);

    header($line);
  }
}


//  fb('Hello World');

//  fb('Log message',FirePHP::LOG);
//  fb('Info message',FirePHP::INFO);
//  fb('Warn message',FirePHP::WARN);
//  fb('Error message',FirePHP::ERROR);

} else {

  fb($data);

  fb($data,'$_SERVER',FirePHP::DUMP);  
}



//header('FirePHP-data: '.substr($data_str,0,strlen($data_str)/2));
//header('FirePHP-data-1: '.substr($data_str,strlen($data_str)/2));

//header('X-FirePHP-ProcessorURL: http://'.$_SERVER['HTTP_HOST'].substr($_SERVER['SCRIPT_NAME'],0,-4).'-Processor.js');

//header('X-FirePHP-RendererURL: http://'.$_SERVER['HTTP_HOST'].substr($_SERVER['SCRIPT_NAME'],0,-4).'-Renderer.js');


require_once('TVarDumper.php');

print TVarDumper::dump($data, 10, true);

var_dump($data);

?>

<script type="application/x-javascript" src="jquery.js"></script>

<p><a href="#" onClick="$.get('Version1.php?var=1');">Version1.php?var=1</a></p>
<p><a href="#" onClick="$.get('ConsoleTest.php');">ConsoleTest.php</a></p>
<p><a href="#" onClick="$.get('ZendTest.php');">ZendTest.php</a></p>
